language/evm/move-to-yul/tests/Arithm.exp

```diff
@@ -0,0 +1,447 @@
+                $result := $t2
+            }
+
+            function A2_M_overflow_u128_mul_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:4774:4775
+                $t2 := $MulU128(x, y)
+                /// @src 2:4772:4777
+                $result := $t2
+            }
+
+            function A2_M_overflow_u64_add(x, y) -> $result {
+                let $t2
+                /// @src 2:3450:3451
+                $t2 := $AddU64(x, y)
+                /// @src 2:3448:3453
+                $result := $t2
+            }
+
+            function A2_M_overflow_u64_add_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:3279:3280
+                $t2 := $AddU64(x, y)
+                /// @src 2:3277:3282
+                $result := $t2
+            }
+
+            function A2_M_overflow_u64_mul(x, y) -> $result {
+                let $t2
+                /// @src 2:4590:4591
+                $t2 := $MulU64(x, y)
+                /// @src 2:4588:4593
+                $result := $t2
+            }
+
+            function A2_M_overflow_u64_mul_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:4436:4437
+                $t2 := $MulU64(x, y)
+                /// @src 2:4434:4439
+                $result := $t2
+            }
+
+            function A2_M_overflow_u8_add(x, y) -> $result {
+                let $t2
+                /// @src 2:3101:3102
+                $t2 := $AddU8(x, y)
+                /// @src 2:3099:3104
+                $result := $t2
+            }
+
+            function A2_M_overflow_u8_add_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:2935:2936
+                $t2 := $AddU8(x, y)
+                /// @src 2:2933:2938
+                $result := $t2
+            }
+
+            function A2_M_overflow_u8_mul(x, y) -> $result {
+                let $t2
+                /// @src 2:4258:4259
+                $t2 := $MulU8(x, y)
+                /// @src 2:4256:4261
+                $result := $t2
+            }
+
+            function A2_M_overflow_u8_mul_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:4092:4093
+                $t2 := $MulU8(x, y)
+                /// @src 2:4090:4095
+                $result := $t2
+            }
+
+            function A2_M_underflow() -> $result {
+                let x, $t1, $t2, $t3
+                $t1 := 0
+                $t2 := 1
+                /// @src 2:2177:2178
+                $t3 := $Sub($t1, $t2)
+                /// @src 2:2175:2180
+                $result := $t3
+            }
+
+            function A2_M_g(x) -> $result {
+                let $t1, $t2
+                $t1 := 2
+                /// @src 2:1689:1690
+                $t2 := $AddU64(x, $t1)
+                /// @src 2:1688:1691
+                $result := $t2
+            }
+
+            function A2_M_f(x) -> $result {
```

Comment:
There are two A2_M_f(x) in the generated code

Reply:
Done